recipes-extended/xen/files/libxl-disable-json-updates.patch

```diff
@@ -1,88 +0,0 @@
-################################################################################
```

Comment:
I think the point of this patch was to not let libxl modify the ```/tmp/xenmgr-xl-<uuid>``` conf file that gets passed as the ```xl create``` arg.  Xenmgr is the authoritative source for what should be in those files; every time a domain is started, xenmgr will overwrite whatever is there with fresh values read from the db.

Reply:
The `/tmp/xenmgr-xl-$uuid` files are different from the json in `/var/lib/xen/userdata-d.$domid.$uuid.json`.  xenmgr still overwites the xl.cfg files and that is fine.  I'm not sure about libxl re-writing the xl.cfg files.  There is some talk of it in the xl man page, but I don't know about it happening in practice.  Xenmgr would still overwrite anyway.

The json data is something libxl uses for its dynamic state tracking.  I am not totally sure what it is for, but there is some logic for comparing it with xenstore when outputting devices.  That leads to some weirdness with `xl pci-list` where it doesn't list any devices for the HVM, but it does list them for the PV stubdom.  The Xenstore entries are only present for the PV stubdom.

Anyway, the patch is affecting the json side and not the xl.cfg.

Xen commit e6afc7063afe06f4609fb2d18fde28f3b54e995e seems to be the switch to the .json stuff.  Looks like that is basically about changes taking affect on reboot.  If we pursue my changes to switch out xenmgr reboot handling to shutdown & start, then this libxl stuff is irrelevant.

Back to this patch.  The description implies failure because the json files are not present.  But they are present now.  I guess I can try switching networks at run time to see that hasn't broken.  Otherwise, I haven't seen issue with this patch dropped.

Reply:
For a running VM, the UI network tab doesn't allow adding or removing networks.  So I'm doubly confused with what this is supposed to be addressing.

Reply:
> For a running VM, the UI network tab doesn't allow adding or removing networks. So I'm doubly confused with what this is supposed to be addressing.

Interesting.  Possibly it was done through the commandline instead of the UI with ```xl network-attach```?  Also another case could be when the ndvm gets restarted underneath a running VM there is some vif switching due to the domid of the ndvm changing.

Reply:
```xl network-list <guest domid>``` should show the updated backend though, and the network in the guest should come back up after the ndvm finishes rebooting.  Xl wouldn't use the ```/tmp/xenmgr-xl``` values until it a new ```xl create``` process is run, or the exiting ```xl create``` does a ```goto``` back to the top where it re-parses the config options (but I think that only happens on a domain reboot, during which xenmgr would re-write the ```/tmp/xenmgr-xl``` file).

Reply:
I tried to differentiate `xec-vm reboot` and guest-internal reboot.  `xec-vm reboot` definitely re-writes the config, and it is available for re-parsing.  For guest-internal, it looks like the XenMgr synchronization signal delays the "goto" until after the xl config is re-written.

network-daemon has moveToNetwork which calls xl network-attach/detach which does the re-attach for dom0 when an NDVM reboots.  Looks like xenmgr calls xl network-detach/attach for other domUs when an NDVM reboots.

xl network-list output looks shows the backend changing.

Using xl network-attach/detach should do the right thing with respect to the json since it's going through libxl.

So what would you like to see with respect to this change?

Reply:
Hmm, I guess if all the edge cases we discussed are working correctly without this patch and we don't see any failures, then it's fine to drop it.